integration_tests/tests/mod.rs

```diff
@@ -0,0 +1,408 @@
+                Some(closing_balances) => closing_balances,
+            };
+
+            // check Customer's version of closing balances
+            if customer_channel.closing_balances().customer_balance != expected_customer_closing
+                || customer_channel.closing_balances().merchant_balance != expected_merchant_closing
+            {
+                return Err(TestError::InvalidClosingBalances {
+                    op: *op,
+                    party: "Customer".to_string(),
+                    expected_customer: expected_customer_closing,
+                    expected_merchant: expected_merchant_closing,
+                    actual_customer: customer_channel.closing_balances().customer_balance,
+                    actual_merchant: merchant_channel.closing_balances().merchant_balance,
+                }
+                .into());
+            }
+
+            // check Merchant's version of closing balances
+            if merchant_channel.closing_balances().customer_balance != expected_customer_closing
+                || merchant_channel.closing_balances().merchant_balance != expected_merchant_closing
+            {
+                return Err(TestError::InvalidClosingBalances {
+                    op: *op,
+                    party: "Merchant".to_string(),
+                    expected_customer: expected_customer_closing,
+                    expected_merchant: expected_merchant_closing,
+                    actual_customer: customer_channel.closing_balances().customer_balance,
+                    actual_merchant: merchant_channel.closing_balances().merchant_balance,
+                }
+                .into());
+            }
+        }
+
+        Ok(())
+    }
+}
+
+#[derive(Debug, Error)]
+#[allow(unused)]
+pub enum LogError {
+    #[error("Failed to open log file: {0}")]
+    OpenFailed(std::io::Error),
+    #[error("Failed to read contents of file: {0}")]
+    ReadFailed(std::io::Error),
+}
+
+#[allow(unused)]
+#[derive(Debug, Clone, Copy)]
+pub enum LogType {
+    Info,
+    Error,
+    Warn,
+}
+
+#[allow(unused)]
+impl LogType {
+    pub fn to_str(self) -> &'static str {
+        match self {
+            LogType::Info => "INFO",
+            LogType::Error => "ERROR",
+            LogType::Warn => "WARN",
+        }
+    }
+}
+
+/// Get any errors from the log file, filtered by party and log type.
+pub fn get_logs(log_type: LogType, party: Party) -> Result<String, LogError> {
+    let mut file = File::open(ERROR_FILENAME).map_err(LogError::OpenFailed)?;
+    let mut logs = String::new();
+    file.read_to_string(&mut logs)
+        .map_err(LogError::ReadFailed)?;
+
+    Ok(logs
+        .lines()
+        .filter(|s| s.contains("zeekoe::"))
+        .filter(|s| s.contains(log_type.to_str()))
+        .filter(|s| s.contains(party.to_str()))
+        .fold("".to_string(), |acc, s| format!("{}{}\n", acc, s)))
+}
+
+/// Helper function to convert human XTZ amount into a `CustomerBalance`.
```

Comment:
todo: Update docs (here and for `to_merchant_balance`) to just describe the action of the function e.g.
```
/// Convert human XTZ amount into a `CustomerBalance`.
```